pkg/sqlutil/test_data_source.go

```diff
@@ -0,0 +1,85 @@
+package sqlutil
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/proullon/ramsql/driver"
```

Comment:
@pavel-raykov did you consider this library?

Reply:
Ramsql does not claim to be compatible with postgres. For example it cannot create the following table:

"CREATE TABLE test (tstamp TIMESTAMP DEFAULT CURRENT_TIMESTAMP)"

with the error 	"Syntax error near default CURRENT_TIMESTAMP" (this for example immediately renders it incompatible with goose that uses CURRENT_TIMESTAMP in the goose_versions table in the db to keep track of the migrations).

If we want a tiny lightweight in-memory db which can closely mock postgres I would propose to use duckdb. (One more advantage of it is that you can start it by typing duckdb in console as opposed to ramsql that does not have a standalone version.)

Reply:
ramsql does come with a standalone version actually:
```
[MB-TQH59VKW66:~/workspace/chainlink-common> ramsql
ramsql> CREATE TABLE foo (a INTEGER);
Query OK. 1 rows affected
ramsql>
```
I'll take a look at duckdb, if that's more compatible with postgres thanks

Reply:
And also, both ramSQL and duckdb are incompatible with goose, so this is one more reason not use them. (Unfortunately, I don't know of any other in-memory db that can be used  [cockroach is better, but it also cannot execute all our migrations =( ])